rules/mapping.go

```diff
@@ -160,6 +131,40 @@ func (mrs *mappingRuleSnapshot) Schema() (*schema.MappingRuleSnapshot, error) {
 	return res, nil
 }
 
+// MappingRuleView is a human friendly representation of a mapping rule at a given point in time.
+type MappingRuleView struct {
```

Comment:
are you wrapping the struct access w/ methods to ensure the fields are immutable?

Reply:
Yes, but I kind of rather just make all the fields exposed. They can't be used for anything.